cli/src/commands/branch.rs

```diff
@@ -298,6 +317,39 @@ fn cmd_branch_create(
     Ok(())
 }
 
+fn cmd_branch_rename(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &BranchRenameArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let target_commit =
+        workspace_command.resolve_single_rev(args.revision.as_deref().unwrap_or("@"), ui)?;
```

Comment:
I think we should remove the `--revision` argument and instead use whatever the branch currently points to. You can get that from `repo.view().get_local_branch(&old_branch)`. That'll give you a `RefTarget` which you can then just pass into `set_local_branch_target()`. Just check that's `is_present()` first, so we give a proper error message if the branch doesn't exist.

Reply:
Done. I have to clone the target commit when passing to set_local_branch_target() to break the ownership by the workspace. Is that the right way to do this?

Reply:
> I have to clone the target commit when passing to set_local_branch_target()

Correct. It's unavoidable anyway since we're going to update the map containing the `&RefTarget`.

Reply:
Cool. Thanks.